extensions/amp-analytics/0.1/test/vendor-requests.json

```diff
@@ -45,6 +45,11 @@
     "trackingIdentifier": "https://!tenantId/s/!pixelId.gif?property=!property&!property=_client_id(cid-scope-cookie-fallback-name)_!extraTrackingValues",
     "defaultTrackingValues": "https://!tenantId/s/!pixelId.gif?property=!property&!property=_client_id(cid-scope-cookie-fallback-name)_!extraTrackingValues&sourceUrl=_source_url_&canonicalUrl=_canonical_url_&referrer=_document_referrer_&title=_title_&visitDate=_timestamp_&geoLocation=_amp_geo(ISOCountry)_&userAgent=_user_agent_"
   },
+  "bluetriangle": {
+    "host": "https://d.btttag.com",
+    "endpoint": "https://d.btttag.com/btt.gif",
+    "pageview": "https://d.btttag.com/btt.gif?gID=!guid&sID=!sessionID&siteID=!prefix&pageName=!pageName&txnName=!trafficSegment&thisURL=_canonical_url_&RefURL=_document_referrer_&dom=0&domContentLoadedEnd=_nav_timing(domContentLoadedEventEnd)_&domContentLoadedStart=_nav_timing(domContentLoadedEventStart)_&domInteractive=_nav_timing(domInteractive)_&domLoading=_nav_timing(domLoading)_&domainLookupStart=_nav_timing(domainLookupStart)_&elementCount=0&fByte=0&fCPnt=_first_contentful_paint_&fetchStart=_nav_timing(fetchStart)_&firstPaint=0&fmp=0&fps=0&fps_tti=0&loadEventStart=_nav_timing(loadEventStart)_&lt_tti=0&nst=0&orderTND=0&pSz=0&pWSz=0&pageValue=&pgTm=0&redirectCount=_nav_redirect_count_&redirectStart=_nav_timing(redirectStart)_&requestStart=_nav_timing(requestStart)_&responseStart=_nav_timing(responseStart)_&secureConnectionStart=_nav_timing(secureConnectionStart)_&ssl=0&tcp=0&top=0&totRD=0&tti=0&tul=0&unloadEventStart=_nav_timing(unloadEventStart)_&wcd=0&eT=7&navigationType=_nav_type_"
```

Comment:
The expected output should be `https://d.btttag.com/btt.gif?gID=_client_id(bluetriangle_guid)_&sID=_client_id(bluetriangle_sessionid)_&siteID=!siteID&pageName=!pageName&txnName=!trafficSegment&thisURL=_canonical_url_&RefURL=_document_referrer_&dom=_nav_timing(domLoading%2CdomComplete)_&domContentLoadedEnd=_nav_timing(domContentLoadedEventEnd)_&domContentLoadedStart=_nav_timing(domContentLoadedEventStart)_&domInteractive=_nav_timing(domInteractive)_&domLoading=_nav_timing(domLoading)_&domainLookupStart=_nav_timing(domainLookupStart)_&elementCount=0&fByte=_nav_timing(requestStart%2CresponseStart)_&fCPnt=FIRST_CONTENTFUL_PAINT&fetchStart=_nav_timing(fetchStart)_&firstPaint=0&fmp=0&fps=0&fps_tti=0&loadEventStart=_nav_timing(loadEventStart)_&lt_tti=0&nst=_nav_timing(navigationStart)_&orderTND=0&pSz=0&pWSz=0&pageValue=&pgTm=_nav_timing(navigationStart%2CloadEventStart)_&redirectCount=_nav_redirect_count_&redirectStart=_nav_timing(redirectStart)_&requestStart=_nav_timing(requestStart)_&responseStart=_nav_timing(responseStart)_&secureConnectionStart=_nav_timing(secureConnectionStart)_&ssl=_nav_timing(secureConnectionStart%2CresponseStart)_&tcp=_nav_timing(connectStart%2CconnectEnd)_&top=0&totRD=0&tti=0&tul=0&unloadEventStart=_nav_timing(unloadEventStart)_&largestContentfulPaint=LARGEST_CONTENTFUL_PAINT&cumulativeLayoutShift=CUMULATIVE_LAYOUT_SHIFT&firstInputDelay=FIRST_INPUT_DELAY&wcd=0&eT=1&navigationType=1`

based on what's defined in `extensions/amp-analytics/0.1/vendors/bluetriangle.json`. Please ensure that this looks good and then change. (i.e. note how you have defined`"guid": "${clientId(bluetriangle_guid)}",`, and so in the request we should have `gID=_client_id(bluetriangle_guid)`).

Reply:
Thanks for fixing the first 2 variables (guid and sID), but there are a lot more fixes that need to happen:

Please just put `https://d.btttag.com/btt.gif?gID=_client_id(bluetriangle_guid)_&sID=_client_id(bluetriangle_sessionid)_&siteID=!siteID&pageName=!pageName&txnName=!trafficSegment&thisURL=_canonical_url_&RefURL=_document_referrer_&dom=_nav_timing(domLoading%2CdomComplete)_&domContentLoadedEnd=_nav_timing(domContentLoadedEventEnd)_&domContentLoadedStart=_nav_timing(domContentLoadedEventStart)_&domInteractive=_nav_timing(domInteractive)_&domLoading=_nav_timing(domLoading)_&domainLookupStart=_nav_timing(domainLookupStart)_&elementCount=0&fByte=_nav_timing(requestStart%2CresponseStart)_&fCPnt=FIRST_CONTENTFUL_PAINT&fetchStart=_nav_timing(fetchStart)_&firstPaint=0&fmp=0&fps=0&fps_tti=0&loadEventStart=_nav_timing(loadEventStart)_&lt_tti=0&nst=_nav_timing(navigationStart)_&orderTND=0&pSz=0&pWSz=0&pageValue=&pgTm=_nav_timing(navigationStart%2CloadEventStart)_&redirectCount=_nav_redirect_count_&redirectStart=_nav_timing(redirectStart)_&requestStart=_nav_timing(requestStart)_&responseStart=_nav_timing(responseStart)_&secureConnectionStart=_nav_timing(secureConnectionStart)_&ssl=_nav_timing(secureConnectionStart%2CresponseStart)_&tcp=_nav_timing(connectStart%2CconnectEnd)_&top=0&totRD=0&tti=0&tul=0&unloadEventStart=_nav_timing(unloadEventStart)_&largestContentfulPaint=LARGEST_CONTENTFUL_PAINT&cumulativeLayoutShift=CUMULATIVE_LAYOUT_SHIFT&firstInputDelay=FIRST_INPUT_DELAY&wcd=0&eT=1&navigationType=1` as the `pageview ` request in `extensions/amp-analytics/0.1/test/vendor-requests.json`

Reply:
FYI you can run unit tests via `amp unit --local_changes`, so we don't have to keep ping-ponging on these fixes.

Reply:
I got this message when running that command :
INFO: Skipping tests on local changes because this is a large refactor.

thoughts?

Reply:
try `amp unit --files=extensions/amp-analytics/0.1/test/test-vendors.js`

Reply:
Finished in 1 min 6.626 secs / 14.521 secs @ 15:48:07 GMT-0400 (Eastern Daylight Time)

SUMMARY:
● 176 tests completed
[15:48:08] Stopped server at http://localhost:8081
[15:48:08] Finished 'unit' after 1m 44s


Looks like the unit test went through successfully